src/include/random.h

```diff
@@ -123,7 +123,7 @@ void randomize_rng()
 template <typename EngineT>
 EngineT& engine()
 {
-  thread_local boost::optional<EngineT> rng_engine;
+  thread_local std::optional<EngineT> rng_engine;
```

Comment:
I've been struggling for hours to make ceph-mon work. Now that it appears to work, rados suddenly doesn't start anymore because "lib/librados.so.2: undefined symbol: rados_create_with_context" ... Sorry, my competence is apparently not enough to run this test.
There are lots of unmerged PRs from me to fix various build failures, and I still stumbler over more build failures every time I try to build Ceph. This frustrates me a lot, and I don't know how you guys manage to run Ceph at all. It's a fragile mess.
In any case, whatever the problem was, it must have been a toolchain bug. Unfortunately, 6 years ago, nobody dug deep enough to find out, so we still don't know.

Reply:
ceph is indeed a beast that grows and grows. we lost some key contributors over the years that kept things sane. it's been hard to find devs willing to go back and clean things up, so i sincerely appreciate your efforts here

i built this branch with gcc-14 and each rados command generated difference nonces, so at least that toolchain works. i'll go ahead and approve, and we'll rely on the rados suite to validate it